File: core/src/main/java/javabot/operations/time/TimezonesEcuador.java
package javabot.operations.time;

public class TimezonesEcuador implements Timezones {

    private static Tri<String> timezones = new Tri<String>( );

    static {
        for( Timezones timezone: Timezones.values( ) ) {
            timezones.insert( timezone.name( ), timezone.getTimezone( ) );
        }
    }

    private enum Timezones {
        Galapagos( "Pacific/Galapagos" ),
        Guayaquil( "America/Guayaquil" );

        private String timezone;

        Timezones( String timezone ) {
            this.timezone = timezone;
        }

        public String getTimezone( ) {
            return timezone;
        }
    }

    @Override
    public String get( String province ) {
        return timezones.get( province );
    }

    @Override
    public String getCapital( ) {
        return Timezones.Guayaquil.getTimezone( );
    }
}

File: src/main/kotlin/javabot/operations/time/TimezonesEcuador.kt
package javabot.operations.time

class TimezonesEcuador : Timezones {

    private enum class Timezones(val timezone: String) {
        Galapagos("Pacific/Galapagos"),
        Guayaquil("America/Guayaquil")
    }

    override fun get(province: String): String? {
        return timezones.get(province)
    }

    override fun getCapital(): String {
        return Timezones.Guayaquil.timezone
    }

    companion object {

        private val timezones = Tri<String>()

        init {
            for (timezone in Timezones.values()) {
                timezones.insert(timezone.name, timezone.timezone)
            }
        }
    }
}
